File: backend/app/schemas/cost.py
from datetime import datetime
from pydantic import BaseModel


class CostSummaryRead(BaseModel):
    total_cost_usd: float
    today_cost_usd: float
    month_cost_usd: float
    total_tokens: int
    total_calls: int
    avg_latency_ms: float


class DailyCostItem(BaseModel):
    date: str
    prompt_tokens: int
    completion_tokens: int
    total_tokens: int
    cost_usd: float
    call_count: int


class ModelCostItem(BaseModel):
    model_name: str
    provider: str
    cost_usd: float
    call_count: int
    token_count: int
    percentage: float


class TopTaskCostItem(BaseModel):
    task_id: int
    task_title: str
    status: str
    cost_usd: float
    total_tokens: int
    model_call_count: int
    created_at: datetime

File: frontend/src/types/cost.ts
export interface CostSummary {
  total_cost_usd: number
  today_cost_usd: number
  month_cost_usd: number
  total_tokens: number
  total_calls: number
  avg_latency_ms: number
}

export interface DailyCostItem {
  date: string
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  cost_usd: number
  call_count: number
}

export interface ModelCostItem {
  model_name: string
  provider: string
  cost_usd: number
  call_count: number
  token_count: number
  percentage: number
}

export interface TopTaskCostItem {
  task_id: number
  task_title: string
  status: string
  cost_usd: number
  total_tokens: number
  model_call_count: number
  created_at: string
}
